datastore/importers/nmdb-import-arista-show-lldp-neighbor/Parser.cpp

```diff
@@ -0,0 +1,340 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+// =============================================================================
+// Maintained by Sandia National Laboratories <Netmeld@sandia.gov>
+// =============================================================================
+
+#include <netmeld/core/utils/StringUtilities.hpp>
+
+#include "Parser.hpp"
+
+namespace nmcu = netmeld::core::utils;
+
+// =============================================================================
+// Parser logic
+// =============================================================================
+Parser::Parser() : Parser::base_type(start)
+{
+  start =
+    +(( noDetailConfig
+      | detailConfig
+      ) [(qi::_val = pnx::bind(&Parser::getData, this))]
+    | ignoredLine
+    )
+    ;
+
+  // ----- no detail -----
+
+  noDetailConfig =
+    noDetailTableInfo
+    >> noDetailHeader
+    > *noDetailEntry [(pnx::bind(&Parser::finalizeData, this))]
+    > *qi::eol
+    ;
+
+  noDetailTableInfo = 
+    qi::lit("Last table change time")
+    > *(!qi::lit("Port") >> ignoredLine)
+    ;
+
+  noDetailHeader =
+    qi::lit("Port")
+    > qi::lit("Neighbor Device ID")  
+    > qi::lit("Neighbor Port ID")
+    > qi::lit("TTL")
+    > qi::eol
+    > +qi::char_('-')
+    > qi::eol
+    ;
+
+  noDetailEntry =
+    token
+    > token [(pnx::ref(nd.curHostname) = qi::_1)]
+    > port  [(pnx::ref(nd.curIfaceName) = qi::_1)]
+    > qi::uint_
+    > qi::eol
+    ;
+
+  // ----- detail -----
+
+  detailConfig =
+    (detailHeader
+    >> (detailEntry | *qi::eol)
+    ) [(pnx::bind(&Parser::finalizeData, this))]
+    ;
+
+  detailHeader =
+    qi::lit("Interface")
+    >> port [(pnx::ref(nd.srcIfaceName) = qi::_1)]
+    >> qi::lit("detected")
+    > ignoredLine
+    > qi::eol
+    ;
+
+  detailEntry =
+    detailNeighborLine
+    > detailDiscovered
+    > +(detailChassisId
+      | detailPortId
+      | detailPortDescription
+      | detailSystemName
+      | detailSystemDescription
+      | detailCapabilities
+      | detailVlan
+      | (!( detailHeader
+          | detailNeighborLine
+          ) >> ignoredLine - qi::eol
+        )
+      )
+    ;
+
+  detailNeighborLine =
+    qi::lit("Neighbor")
+    > +(qi::char_ - '/')
+    > '/'
+    > ( inQuotes | port)
+    > qi::lit(", age")
+    > ignoredLine
+    ;
+
+  detailDiscovered =
+    qi::lit("Discovered")
+    > qi::uint_
+    > qi::lit("days,")
+    > token
+    > ignoredLine
+    ;
```

Comment:
I'm wondering if this isn't too specific (i.e., I'm not sure what if it would say "0 days" or just not put anything).  I don't think there are any other lines that start with "Discovered" and the rule is only used in a specific context, so we perhaps should just ignore after that.  Thoughts?